examples/cli/src/main/java/io/unitycatalog/cli/delta/DeltaKernelUtils.java

```diff
@@ -81,30 +86,33 @@ public static FileSystem getFileSystem(URI tablePathURI, Configuration conf) {
   }
 
   public static Configuration getHDFSConfiguration(
-      URI tablePathUri, AwsCredentials awsTempCredentials) {
+      URI tablePathUri, TemporaryCredentials temporaryCredentials) {
     Configuration conf = new Configuration();
-    if (tablePathUri.getScheme() != null
-        && tablePathUri.getScheme().equals("s3")
-        && awsTempCredentials == null) {
-      throw new IllegalArgumentException("AWS temporary credentials are missing");
+    String scheme = tablePathUri.getScheme();
+    if (scheme == null) {
+      throw new IllegalArgumentException("URI scheme is missing");
     }
-    if (tablePathUri.getScheme().equals("s3")) {
+    if (scheme.equals(Constants.URI_SCHEME_S3)) {
+      AwsCredentials awsTempCredentials = temporaryCredentials.getAwsTempCredentials();
+      if (awsTempCredentials == null) {
+        throw new IllegalArgumentException("AWS temporary credentials are missing");
+      }
       conf.set("fs.s3a.access.key", awsTempCredentials.getAccessKeyId());
       conf.set("fs.s3a.secret.key", awsTempCredentials.getSecretAccessKey());
       conf.set("fs.s3a.session.token", awsTempCredentials.getSessionToken());
       conf.set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem");
       conf.set("fs.s3a.path.style.access", "true");
```

Comment:
Could you please also file a separate issue, so that we can reuse the credential properties setting with the https://github.com/unitycatalog/unitycatalog/blob/fc76ab52aa03550d6577eca7be90004a5d2913b8/connectors/spark/src/main/scala/io/unitycatalog/spark/auth/CredPropsUtil.java#L19

I think the credentials properties setting  should be same.